recipes/gcc/all/test_package/conanfile.py

```diff
@@ -1,34 +1,81 @@
-from conans import ConanFile, tools
 import os
+import shutil
+from conan import ConanFile
+from conan.tools.build import cross_building
+from conan.tools.env import VirtualBuildEnv, VirtualRunEnv
 
 
 class TestPackageConan(ConanFile):
     settings = "os", "compiler", "build_type", "arch"
-    generators = "cmake"
+
+    @property
+    def file_io(self):
+        return {
+            "c": {
+                "compiler": "$CC",
+                "src": os.path.join(self.source_folder, "hello.c"),
+                "bin": os.path.join(self.build_folder, "hello_c"),
+            },
+            "cpp": {
+                "compiler": "$CXX",
+                "src": os.path.join(self.source_folder, "hello.cpp"),
+                "bin": os.path.join(self.build_folder, "hello_cpp"),
+            },
+            "fortran": {
+                "compiler": "$FC",
+                "src": os.path.join(self.source_folder, "hello.f90"),
+                "bin": os.path.join(self.build_folder, "hello_f90"),
+            },
+        }
+
+    def generate(self):
```

Comment:
I don't believe you need this generate method here but I found that I needed to add the following classic requirements() method in order for the tests to work on Conan 2.0:

>     def requirements(self):
>         self.requires(self.tested_reference_str)

Reply:
The concern that I have around removing the `generate` method, is that invokes a user requirement to have the following in their profile:

```
[conf]
tools.env.virtualenv.auto_use=True
```
which will be fine for cci, but won't be obvious if anybody tries to build for themselves on 1.X. I think my preference would be to retain the `generate()` method until the above conf is not required